Add a few missing endpoints and add some missing types

YummyBacon5 opened the pull request:
This adds missing endpoints:
- `GetStaticData`
- `PutUserUpdateEmail`
- `PutUserUpdateName`
- `PutGameFollowerOrder`
- `PutThemeSettings`
- `GetUserApiKey`
- `GetUserGameBoostData`
- `GetUserDataExport`
- `GetUserComments`

This also adds some missing keyword arguments. Most notably to `GetGameLeaderboard2`, `GetGameLeaderboard`, `GetGameLevelSummary`, and others.
This also fixes some types.

YummyBacon5 committed:
Add a few missing endpoints and add some missing types

YummyBacon5 committed:
Fix incorrect name

YummyBacon5 committed:
Move endpoints to more correct locations

ManicJamie commented on src/speedruncompy/datatypes/enums.py:
```diff
@@ -60,9 +60,9 @@ class ObsoleteFilter(IntEnum):
 
 class VerifiedFilter(IntEnum):
-    OPTIONAL = 0
+    AWAITING = 0 # Speedrun.com calls it "Awaiting" in its code on this enum
```
Not sure I'm a fan of this particular name change; SRC may call it that internally but "Awaiting" means specifically "Not Verified" to me, not "May be verified or not"

YummyBacon5 replied:
Awaiting just means pending. I could rename it to pending, but then the `Verified` enum could be used.

ManicJamie commented on README.md:
```diff
@@ -31,18 +28,16 @@ Note that this uses the API in the same way as [https://speedrun.com]. The v2 AP
 
-client = SpeedrunClient("my_app_name")
-client.PHPSESSID = "secret PHPSESSID"  # You should store this separately!
+client = SpeedrunClient("my_app_name", PHPSESSID=os.getenv("PHPSESSID")) # You should store this separately!
```
To be beginner friendly (assuming this lib is being used by disgruntled mods more than programmers :p) it's probably worth noting what os.getenv() does & separate it out for clarity.

```suggestion
secret = os.getenv("PHPSESSID")
# You shouldn't store PHPSESSID in the script directly, instead load it externally.
# `os.getenv()` assumes you have already set env variable PHPSESSID in your terminal;
# you can load it from a file instead using `open()` or library `load_dotenv`.

client = SpeedrunClient("my_app_name", PHPSESSID=secret)
```

ManicJamie commented on src/speedruncompy/datatypes/defs.py:
```diff
@@ -488,12 +489,41 @@ class GameBoost(Datatype):
+class Color(Datatype):
+    id: str
+    darkColor: str
+    """Deprecated, darkColour is always used on the site"""
```
Is this docstring meant to be under `lightColor`?

YummyBacon5 replied:
Yes, as I copied what was elsewhere in the code

ManicJamie commented on src/speedruncompy/endpoints.py:
```diff
@@ -19,19 +19,53 @@ class GetGameLeaderboard2(GetRequest[r_GetGameLeaderboard2], BasePaginatedReques
     - @page
     """
-    def __init__(self, gameId: str, categoryId: str, _api: SpeedrunClient | None = None, **params) -> None:
+    def __init__(
```
I would rather leave out optional keywords in favour of (reformatted to ReST for broader editor support, damn you VSCode) docstrings, since all keyword arguments are passed to the API, which feels like a smell (if a default on SRC changes, the library will not notice it).

However since the work appears to be mostly done already I could be convinced otherwise.

YummyBacon5 replied:
Their default type is None, so, I would have thought that these wouldn't be sent to the API.
Also, a doc string's autocomplete won't be as good as these types. And you have these optional keywords in other parts of this code.

ManicJamie reviewed (changes requested):
Generally fine - I'd like some testing coverage and/or response dumps for some of the datatypes/defs changes, but I haven't kept on top of coverage lately anyway so I can't fault you there.

Some linting done but it's clear I need to run a global linter pass regardless.

ManicJamie committed:
Linting pass

To be joined by a more major linter pass when I next get the time to do major work on this repo

YummyBacon5 committed:
Improve README.md about env

Co-authored-by: Jamie <jamiebloomfield8@gmail.com>

YummyBacon5 committed:
Add GetUserComments

YummyBacon5 committed:
Reuse variable

YummyBacon5 committed:
Update doc string

YummyBacon5 committed:
Add tests and fixes some types

YummyBacon5 committed:
Update the GetUserGameBoostData test and edit the types

YummyBacon5 committed:
Use the correct id

YummyBacon5 committed:
Fix theme settings test

YummyBacon5 commented:
@ManicJamie I've now added tests.

I can't test the tests on the Hornet_Bot account. But the tests do still seem like they'll work

ManicJamie commented:
Has some minor issues but easier for me to fix in-place, otherwise lgtm
